src/main/java/bio/terra/service/profile/flight/update/UpdateProfileMetadataStep.java

```diff
@@ -0,0 +1,68 @@
+package bio.terra.service.profile.flight.update;
+
+import bio.terra.model.BillingProfileModel;
+import bio.terra.model.BillingProfileUpdateModel;
+import bio.terra.service.iam.AuthenticatedUserRequest;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.profile.ProfileService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+public class UpdateProfileMetadataStep implements Step {
+
+    private final ProfileService profileService;
+    private final BillingProfileUpdateModel profileRequest;
+    private final AuthenticatedUserRequest user;
+    private static final Logger logger = LoggerFactory.getLogger(UpdateProfileMetadataStep.class);
+
+    public UpdateProfileMetadataStep(ProfileService profileService,
+                                     BillingProfileUpdateModel profileRequest,
+                                     AuthenticatedUserRequest user) {
+        this.profileService = profileService;
+        this.profileRequest = profileRequest;
+        this.user = user;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext context) {
+        // get current billing model so we know what to revert to
+        BillingProfileModel oldProfileModel = profileService.getProfileById(profileRequest.getId(), user);
+        FlightMap workingMap = context.getWorkingMap();
+        workingMap.put(JobMapKeys.REVERT_TO.getKeyName(), oldProfileModel);
+
+        // TODO: Is there a better way to handle this?
```

Comment:
This seems OK to me but I would only check for `null`, unless you know there's a way an empty string can end up here and everything else is correct.